TheaterSchedule/Controllers/ScheduleController.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System;
+using System.Collections.Generic;
+using Microsoft.AspNetCore.Mvc;
+using TheaterSchedule.BLL;
+using TheaterSchedule.DAL;
+using TheaterSchedule.DAL.Contexts;
+using TheaterSchedule.DAL.Entities;
+using TheaterSchedule.DAL.Repositories;
+
+namespace TheaterSchedule.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class ScheduleController : Controller
+    {
+        IScheduleServices services;
+
+        public ScheduleController(TheaterScheduleContext context)
+        {
+            this.services = new ScheduleServices(new TheaterScheduleUnitOfWork(context));
+        }
+
+        [HttpGet("startDate={startDate}&endDate={endDate}")]
+        public IEnumerable<Schedule> FilterPerformancesByDateRange(string startDate = null, string endDate = null)
```

Comment:
Use nullable datetime parameters
https://stackoverflow.com/questions/48753717/how-to-add-an-action-with-a-datetime-parameter